File: stress/t/arr_prefixsum.c
#include <stdio.h>
#include <stdint.h>
int main(void){
  int64_t i,n,run,t,last;
  int64_t A[25],P[25];
  n=25;
  for(i=0;i<n;i++){ A[i]=(i*7+3)%17; }
  run=0;
  for(i=0;i<n;i++){ t=A[i]; run=run+t; P[i]=run; }
  t=P[0]; printf("p0="); printf("%ld",(long)t); printf("\n");
  t=P[12]; printf("p12="); printf("%ld",(long)t); printf("\n");
  last=P[24]; printf("plast="); printf("%ld",(long)last); printf("\n");
  return 0;
}

File: stress/t/arr_prefixsum.f
      PROGRAM ARRPREFIX
      INTEGER*8 I,N,RUN,T,LAST
      INTEGER*8 A(0:24),P(0:24)
      N=25
      I=0
   10 IF (I.GE.N) GOTO 20
      A(I)=MOD(I*7+3,17_8)
      I=I+1
      GOTO 10
   20 CONTINUE
      RUN=0
      I=0
   30 IF (I.GE.N) GOTO 40
      T=A(I)
      RUN=RUN+T
      P(I)=RUN
      I=I+1
      GOTO 30
   40 CONTINUE
      T=P(0)
      WRITE(*,'(A,I0)') 'p0=',T
      T=P(12)
      WRITE(*,'(A,I0)') 'p12=',T
      LAST=P(24)
      WRITE(*,'(A,I0)') 'plast=',LAST
      END
